schema/annotation-source.yaml

```diff
@@ -67,12 +68,15 @@ $defs:
         const: Contribution
         default: Contribution
         description: MUST be "Contribution".
-      agent:
+      contribution_made_by:
         $ref: "#/$defs/Agent"
-      date:
+      started:
         type: string
         format: date
-      role:
+      ended:
+        type: string
+        format: date
+      contributor_role:
```

Comment:
'role' vs 'contributor_role' . . . additional specificity is useful here as it clarifies that this is the role played by the contributing agent in executing their contribution (vs the role of the artifact they contributed to, or an artifact they may have used in making the contribution). I see no harm in being more specific here.

Reply:
I'll submit here. I believe that `activity` or `contribution_activity`  is the cleanest expression of what we are trying to use this attribute to convey. If we can make sure the description of `contributor_role` explicitly says that this is the role that reflects the activity of the contribution being made by the contributor then I'm fine with leaving the term `activity` out of the attribute name. 

I also believe that if we follow with the logic that our attribute names reflect the entity they hang off of then this attribute should be embedded within the `contributor` (agent) - which makes little sense, since it is about the contribution activity that the contributor is performing. If we do leave it under `contribution` then shouldn't it be `contribution_role` ?

Reply:
This attribute captures the type of role played by the agent in making their contribution - e.g. for a publication, that they played an 'editor' role, or a 'figure creator' role, or a 'primary author' role.  As you point out, this could also be framed as the type of contribution activity performed (e.g. "contribution_type": "figure creation") - but I am inclined to treat these as roles so we can leverage existing ontologies / frameworks like the Contributor Role Ontology.

To me `contribution_role` is less clear - as other things may make play roles in a Contribution activity (e.g. PaperPile software playing the role of 'citation manager' in an agents contribution to authoring publication). Also, `contribution_role` could be construed as describing a role that the contribution itself played.  By contrast, naming this 'contributor_role' IMO makes it clear that we are talking about the role played _by the contributor_, in making their contribution.

Reply:
So, all things above considered re: naming of Contribution attributes - I'll put forth the following proposal:
- **id**: string
- **contributor**: Agent
- **contributor_role**: Coding
- **date**: dateTime